### ISSUE TITLE
BigQuery client using wrong project id when creating session

### ISSUE THREAD

**Issue by regadas** [trinodb/trino#15867]
Hi all,

Updating from `403` to `405` when using BigQuery connector we are faced with the following error:

```
Query 20230124_184824_00005_h8n4u failed: io.grpc.StatusRuntimeException: PERMISSION_DENIED: request failed: the user does not have 'bigquery.readsessions.create' permission for 'projects/foobar'
```

This means that the project set in the BigQuery connector properties is being used as request project instead of the parent project.

If I'm not mistaken, this issue is due to https://github.com/trinodb/trino/pull/14961

**Comment by hashhar** [comment]
So if I understand correctly the order of project id being chosen should be following:

- Parent project id
- Catalog project id
- Credentials project id

The fix in #14961 was meant to fix issue where you couldn't actually use two different bigquery catalogs if you used same set of credentials but I think you're right that the new behaviour is still buggy but in a different way.

**Comment by hashhar** [comment]
Maybe while we're doing this we should rename "parent" to something more obvious?
